src/main/java/org/folio/innreach/domain/service/impl/ItemHoldServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+import org.folio.innreach.domain.service.ItemHoldService;
+import org.folio.innreach.domain.service.ItemService;
+import org.folio.innreach.domain.service.MaterialTypeMappingService;
+import org.folio.innreach.dto.InnReachResponseDTO;
+import org.folio.innreach.dto.TransactionHoldDTO;
+import org.folio.innreach.mapper.InnReachErrorMapper;
+import org.folio.innreach.mapper.InnReachTransactionHoldMapper;
+import org.folio.innreach.repository.InnReachTransactionRepository;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityExistsException;
+import java.util.List;
+
+@Log4j2
+@RequiredArgsConstructor
+@Service
+public class ItemHoldServiceImpl implements ItemHoldService {
+  private final InnReachTransactionRepository transactionRepository;
+  private final InnReachTransactionHoldMapper transactionHoldMapper;
+  private final ItemService itemService;
+  private final CentralServerService centralServerService;
+  private final MaterialTypeMappingService materialService;
+  private final InnReachErrorMapper errorMapper;
+
+  private InnReachTransaction createTransactionWithItemHold(String trackingId, String centralCode) {
+    var transaction = new InnReachTransaction();
+    transaction.setTrackingId(trackingId);
+    transaction.setCentralServerCode(centralCode);
+    transaction.setType(InnReachTransaction.TransactionType.ITEM);
+    transaction.setState(InnReachTransaction.TransactionState.ITEM_HOLD);
+    return transaction;
+  }
+
+  @Override
+  public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId, String centralCode, TransactionHoldDTO dto) {
+    var response = new InnReachResponseDTO();
+    response.setStatus("ok");
+    try {
+      transactionRepository.fetchOneByTrackingId(trackingId).ifPresent(m -> {
+        throw new EntityExistsException("INN-Reach Transaction with tracking ID = " + trackingId
+          + " already exists.");
+      });
+      var centralServer = centralServerService.getCentralServerByCentralCode(centralCode);
+      var centralServerId = centralServer.getId();
+      var transaction = createTransactionWithItemHold(trackingId, centralCode);
+      var itemHold = transactionHoldMapper.toItemHold(dto);
+      var item = itemService.getItemByHrId(itemHold.getItemId());
+      var materialTypeId = item.getMaterialType().getId();
+      var materialType = materialService.findByCentralServerAndMaterialType(centralServerId, materialTypeId);
+      itemHold.setCentralItemType(materialType.getCentralItemType());
+      transaction.setHold(itemHold);
+      transactionRepository.save(transaction);
+    } catch (Exception e) {
+      log.warn("An error occurred during creation of INN-Reach Transaction.", e);
+      response.setStatus("failed");
+      response.setReason("An error occurred during creation of INN-Reach Transaction.");
+      var innReachError = errorMapper.toInnReachError(e);
+      response.setErrors(List.of(innReachError));
+    }
```

Comment:
lets just throw a `CirculationException(<msg>, e)`; instead. The INN-Reach error then will be prepared by `D2irExceptionHandlerController`